### DISCUSSION TITLE
Failed to fetch (self-signed certificate)

### DISCUSSION THREAD

**Discussion by lpezet** [hpcc-systems/vscode-ecl#272] [Q&A] [answered]
Hello!

I just installed the extension in VSCode (1.58.2) on Mac (BigSur, 11.5). I can compile and run ECL code but it won't display results in the "HPCC: [workunitid]" panel.
Instead if says:
`0: Failed to fetch - 0 Issues, 0 Results.`
I can click on the globe to get to the ECL Watch page in my browser and see everything there (including the expected **Result 1** I'd like to see in VSCode).
I turned on  "ecl.debugLogging" in settings.json (and restarted VSCode) but I don't see any output besides the usual INFO logs in the OUTPUT panel.
My eclcc.log file only mentions loading library files.

I use HTTPS protocol with username and password in my launch configuration (and Http Proxy set to off in global settings).

What am I doing wrong?
Is there a way to get more logs about that "Failed to fetch" problem?

**Comment by GordonSmith** [comment]
@lpezet - What server version are you targetting?

**Comment by lpezet** [comment]
man I tried to provide it all in first post and I forgot that one! Sorry!
Server is 8.2.0-2.

**Comment by GordonSmith** [comment]
Also to see the additional logging you will need to look here:
![image](https://user-images.githubusercontent.com/966863/127349120-e1654614-372d-49bf-aebc-82e9903f1e41.png)

Unfortunately I don't think there will be any additional information there, because the result view is an embedded HTML page within VSCode...

But instead if you select Developer: Open Webview developer Tools:
![image](https://user-images.githubusercontent.com/966863/127349467-bea4a212-668a-41a4-aaf4-f868070e1ee2.png)

And look in network tab for failed fetches (WUResult):
![image](https://user-images.githubusercontent.com/966863/127349889-f20b3c40-d30e-45bc-b804-215253acd51b.png)

**Comment by lpezet** [comment, marked as answer]
Oh ok. I didn't have those DEBUG logs and turned out I need to set "ECL: Debug Logging" in the Extension Settings and NOT in the settings.json (when I mouse over "ecl.debugLogging" in settings.json VSCode ddid warn me "This setting can be applied only in application user settings"). My bad.
I see the DEBUG logs and nothing that helps like you said.
In the Network tab, I see **WUQuery.json** requests that failed. If I copy those requests for like curl, they work fine BUT ONLY if I don't verify the SSL certificate (I'm using a self-signed cert right now).
Looking at the Console tab I see **"Failed to load resource: net::ERR_CERT_INVALID"** for those WQuery.json requests.

Sigh...any way to make it work with self-signed certificates?

**Comment by GordonSmith** [comment]
If you add it to your Chrome / Chromium settings it might get picked up by VSCode as well? 
 (chrome://settings/privacy  ->  Manage Certificates)

**Comment by lpezet** [comment]
So I tried to import my self-signed certificate in Keychain Access, but it doesn't seem do any good (it reads the certificate just fine, but still have the same issues).
Found another way until I figure that one out (or use a CA issued one):
`code --ignore-certificate-errors`

That works for me and now I see my query results in VSCode!

Thank you so @GordonSmith for this super helpful and very responsive support!

**Comment by GordonSmith** [comment]
Good find - the following extension may help as well (for our windows users):  https://marketplace.visualstudio.com/items?itemName=ukoloff.win-ca
